### ISSUE TITLE
stuck in generating !!!!!!!!!!!!!!!!!

### ISSUE THREAD

**Issue by yiyepiaoling0715** [sail-sg/Precision-RL#11]
`{'actor_rollout_ref': {'actor': {'checkpoint': {'load_contents': ['model',
                                                                  'optimizer',
                                                                  'extra'],
                                                'save_contents': ['model',
                                                                  'optimizer',
                                                                  'extra']},
                                 'clip_ratio': 0.2,
                                 'clip_ratio_c': 3.0,
                                 'clip_ratio_high': 0.28,
                                 'clip_ratio_low': 0.2,
                                 'dtype': 'float16',
                                 'entropy_checkpointing': False,
                                 'entropy_coeff': 0,
                                 'entropy_from_logits_with_chunking': True,
                                 'fsdp_config': {'forward_prefetch': False,
                                                 'fsdp_size': -1,
                                                 'model_dtype': 'float16',
                                                 'offload_policy': True,
                                                 'optimizer_offload': False,
                                                 'param_offload': True,
                                                 'reshard_after_forward': True,
                                                 'wrap_policy': {'min_num_params': 0}},
                                 'grad_clip': 1.0,
                                 'kl_loss_coef': 0.001,
                                 'kl_loss_type': 'low_var_kl',
                                 'loss_agg_mode': 'seq-mean-token-sum-norm',
                                 'optim': {'lr': 1e-06,
                                           'lr_warmup_steps': -1,
                                           'lr_warmup_steps_ratio': 0.0,
                                           'min_lr_ratio': 0.0,
                                           'num_cycles': 0.5,
                                           'total_training_steps': -1,
                                           'warmup_style': 'constant',
                                           'weight_decay': 0.01},
                                 'policy_loss': {'clip_cov_lb': 1.0,
                                                 'clip_cov_ratio': 0.0002,
                                                 'clip_cov_ub': 5.0,
                                                 'kl_cov_ratio': 0.0002,
                                                 'loss_mode': 'vanilla',
                                                 'ppo_kl_coef': 0.1},
                                 'ppo_epochs': 1,
                                 'ppo_max_token_len_per_gpu': 129024,
                                 'ppo_micro_batch_size': None,
                                 'ppo_micro_batch_size_per_gpu': 1,
                                 'ppo_mini_batch_size': 8,
                                 'shuffle': False,
                                 'strategy': 'fsdp2',
                                 'ulysses_sequence_parallel_size': 8,
                                 'use_dynamic_bsz': False,
                                 'use_kl_loss': False,
                                 'use_torch_compile': False},
                       'hybrid_engine': True,
                       'model': {'custom_chat_template': None,
                                 'enable_activation_offload': False,
                                 'enable_gradient_checkpointing': True,
                                 'exclude_modules': None,
                                 'external_lib': None,
                                 'fused_kernel_options': {'impl_backend': 'torch'},
                                 'lora_alpha': 16,
                                 'lora_rank': 0,
                                 'override_config': {},
                                 'path': '**/qwen3-14b',
                                 'target_modules': 'all-linear',
                                 'trust_remote_code': False,
                                 'use_fused_kernels': False,
                                 'use_liger': False,
                                 'use_remove_padding': True,
                                 'use_shm': False},
                       'profiler': {'_target_': 'verl.utils.profiler.ProfilerConfig',
                                    'all_ranks': False,
                                    'discrete': False,
                                    'ranks': []},
                       'ref': {'entropy_checkpointing': False,
                               'entropy_from_logits_with_chunking': True,
                               'fsdp_config': {'forward_prefetch': False,
                                               'param_offload': True,
                                               'reshard_after_forward': True,
                                               'wrap_policy': {'min_num_params': 0}},
                               'log_prob_max_token_len_per_gpu': 129024,
                               'log_prob_micro_batch_size': None,
                               'log_prob_micro_batch_size_per_gpu': 1,
                               'log_prob_use_dynamic_bsz': False,
                               'strategy': 'fsdp2',
                               'ulysses_sequence_parallel_size': 8,
                               'use_torch_compile': False},
                       'rollout': {'agent': {'agent_loop_config_path': None,
                                             'custom_async_server': {'name': None,
                                                                     'path': None},
                                             'num_workers': 0},
                                   'calculate_log_probs': False,
                                   'disable_log_stats': True,
                                   'do_sample': True,
                                   'dtype': 'float16',
                                   'enable_chunked_prefill': False,
                                   'enforce_eager': True,
                                   'engine_kwargs': {'sglang': {'attention_backend': None},
                                                     'vllm': {'disable_mm_preprocessor_cache': False,
                                                              'swap_space': None}},
                                   'free_cache_engine': True,
                                   'gpu_memory_utilization': 0.4,
                                   'ignore_eos': False,
                                   'layered_summon': False,
                                   'load_format': 'dummy_dtensor',
                                   'log_prob_max_token_len_per_gpu': 129024,
                                   'log_prob_micro_batch_size': None,
                                   'log_prob_micro_batch_size_per_gpu': 1,
                                   'log_prob_use_dynamic_bsz': False,
                                   'max_model_len': None,
                                   'max_num_batched_tokens': 129024,
                                   'max_num_seqs': 1024,
                                   'mode': 'async',
                                   'multi_stage_wake_up': False,
                                   'multi_turn': {'completion_callback': None,
                                                  'enable': False,
                                                  'format': 'hermes',
                                                  'interaction_config_path': None,
                                                  'max_assistant_turns': None,
                                                  'max_parallel_calls': 1,
                                                  'max_tool_response_length': 256,
                                                  'max_user_turns': None,
                                                  'tokenization_sanity_check_mode': 'strict',
                                                  'tool_config_path': None,
                                                  'tool_response_truncate_side': 'middle',
                                                  'use_inference_chat_template': False},
                                   'n': 4,
                                   'name': 'vllm',
                                   'prompt_length': 40960,
                                   'response_length': 2048,
                                   'temperature': 0.7,
                                   'tensor_model_parallel_size': 8,
                                   'top_k': -1,
                                   'top_p': 1,
                                   'trace': {'backend': None,
                                             'token2text': False},
                                   'update_weights_bucket_megabytes': 512,
                                   'val_kwargs': {'do_sample': True,
                                                  'n': 4,
                                                  'temperature': 0.7,
                                                  'top_k': 20,
                                                  'top_p': 0.8}}},
 'algorithm': {'_target_': 'verl.trainer.config.AlgoConfig',
               'adv_estimator': 'grpo',
               'gamma': 1.0,
               'kl_ctrl': {'_target_': 'verl.trainer.config.KLControlConfig',
                           'horizon': 10000,
                           'kl_coef': 0.001,
                           'target_kl': 0.1,
                           'type': 'fixed'},
               'kl_penalty': 'kl',
               'lam': 1.0,
               'norm_adv_by_std_in_grpo': False,
               'pf_ppo': {'_target_': 'verl.trainer.config.PFPPOConfig',
                          'reweight_method': 'pow',
                          'weight_pow': 2.0},
               'use_kl_in_reward': False,
               'use_pf_ppo': False},
 'critic': {'_target_': 'verl.trainer.config.FSDPCriticConfig',
            'checkpoint': {'load_contents': ['model', 'optimizer', 'extra'],
                           'save_contents': ['model', 'optimizer', 'extra']},
            'cliprange_value': 0.5,
            'forward_max_token_len_per_gpu': 32768,
            'forward_micro_batch_size': None,
            'forward_micro_batch_size_per_gpu': None,
            'grad_clip': 1.0,
            'loss_agg_mode': 'seq-mean-token-sum-norm',
            'model': {'enable_activation_offload': False,
                      'enable_gradient_checkpointing': True,
                      'external_lib': None,
                      'fsdp_config': {'forward_prefetch': False,
                                      'fsdp_size': -1,
                                      'offload_policy': False,
                                      'optimizer_offload': False,
                                      'param_offload': False,
                                      'reshard_after_forward': True,
                                      'wrap_policy': {'min_num_params': 0}},
                      'lora_alpha': 16,
                      'lora_rank': 0,
                      'override_config': {},
                      'path': '~/models/deepseek-llm-7b-chat',
                      'target_modules': 'all-linear',
                      'tokenizer_path': '**/qwen3-14b',
                      'trust_remote_code': False,
                      'use_remove_padding': False,
                      'use_shm': False},
            'optim': {'lr': 1e-05,
                      'lr_warmup_steps_ratio': 0.0,
                      'min_lr_ratio': None,
                      'total_training_steps': -1,
                      'warmup_style': 'constant',
                      'weight_decay': 0.01},
            'ppo_epochs': 1,
            'ppo_max_token_len_per_gpu': 32768,
            'ppo_micro_batch_size': None,
            'ppo_micro_batch_size_per_gpu': None,
            'ppo_mini_batch_size': 8,
            'profiler': {'_target_': 'verl.utils.profiler.ProfilerConfig',
                         'all_ranks': False,
                         'discrete': False,
                         'ranks': []},
            'rollout_n': 4,
            'shuffle': False,
            'strategy': 'fsdp',
            'ulysses_sequence_parallel_size': 1,
            'use_dynamic_bsz': False},
 'custom_reward_function': {'name': 'compute_score', 'path': None},
 'data': {'custom_cls': {'name': None, 'path': None},
          'datagen': {'name': None, 'path': None},
          'dataloader_num_workers': 8,
          'filter_overlong_prompts': False,
          'filter_overlong_prompts_workers': 1,
          'gen_batch_size': 24,
          'image_key': 'images',
          'max_prompt_length': 40960,
          'max_response_length': 2048,
          'prompt_key': 'prompt',
          'return_full_prompt': False,
          'return_multi_modal_inputs': True,
          'return_raw_chat': False,
          'return_raw_input_ids': False,
          'reward_fn_key': 'data_source',
          'sampler': {'class_name': None, 'class_path': None},
          'shuffle': True,
          'tokenizer': None,
          'train_batch_size': 8,
          'train_files': '***/train_verl.parquet',
          'truncation': 'error',
          'trust_remote_code': False,
          'use_shm': False,
          'val_batch_size': 32,
          'val_files': '**/test_verl.parquet',
          'validation_shuffle': False,
          'video_key': 'videos'},
 'fireworks': {'concurrency': 32,
               'deployment_id': None,
               'model_id_prefix': 'test-model'},
 'ray_init': {'num_cpus': None, 'timeline_json_file': None},
 'reward_model': {'enable': False,
                  'forward_max_token_len_per_gpu': 32768,
                  'launch_reward_fn_async': False,
                  'max_length': None,
                  'micro_batch_size': None,
                  'micro_batch_size_per_gpu': None,
                  'model': {'external_lib': None,
                            'fsdp_config': {'forward_prefetch': False,
                                            'fsdp_size': -1,
                                            'param_offload': False,
                                            'reshard_after_forward': True,
                                            'wrap_policy': {'min_num_params': 0}},
                            'input_tokenizer': '**/qwen3-14b',
                            'path': '~/models/FsfairX-LLaMA3-RM-v0.1',
                            'trust_remote_code': False,
                            'use_fused_kernels': False,
                            'use_remove_padding': False,
                            'use_shm': False},
                  'profiler': {'_target_': 'verl.utils.profiler.ProfilerConfig',
                               'all_ranks': False,
                               'discrete': False,
                               'ranks': []},
                  'reward_manager': 'naive',
                  'sandbox_fusion': {'max_concurrent': 64,
                                     'memory_limit_mb': 1024,
                                     'url': None},
                  'strategy': 'fsdp',
                  'ulysses_sequence_parallel_size': 1,
                  'use_dynamic_bsz': False},
 'rllm': {'accumulate_reasoning': True,
          'agent': {'agent_args': {},
                    'engine_args': {},
                    'max_steps': 40,
                    'name': 'math_agent',
                    'overlong_filter': False,
                    'trajectory_timeout': None},
          'apr_repo_dirpath': '***/apr_ruan_pool_121205',
          'bool_debug': False,
          'compact_filtering': {'enable': True,
                                'mask_env_done': False,
                                'mask_error': True,
                                'mask_max_prompt_length_exceeded': True,
                                'mask_max_response_length_exceeded': True,
                                'mask_max_turns_exceeded': False,
                                'mask_timeout': True,
                                'mask_unknown': False},
          'dataset_name': 'apr_ruanan_prod_20251220_2391',
          'debug': {'max_futures': 11,
                    'max_samples_per_batch': 11,
                    'max_trajectory_generations': 15,
                    'max_val_batches': 3},
          'disable_thinking': False,
          'env': {'env_args': {}, 'name': 'custom'},
          'filter_message_type': ['think'],
          'mask_truncated_samples': False,
          'max_memory_tokens': 40860,
          'rejection_sample': {'enable': True, 'multiplier': 3},
          'stepwise_advantage': {'enable': False,
                                 'mode': 'broadcast',
                                 'normalize_by_steps': False},
          'workflow': {'n_parallel_tasks': 768,
                       'name': 'single_turn_workflow',
                       'raise_on_error': True,
                       'retry_limit': 1,
                       'use_workflow': True,
                       'workflow_args': {'agent_args': {},
                                         'agent_cls': None,
                                         'env_args': {},
                                         'env_cls': None,
                                         'env_save_jsonpath': '**',
                                         'gamma': 0.0,
                                         'log_base_dir': '/lpai/output/logs',
                                         'reward_bonus_coeff': 0.0,
                                         'reward_score_of_termination_error': -1.0,
                                         'sopagent_workflow_name': 'apr_ruanan',
                                         'timeout': 18000}}},
 'trainer': {'balance_batch': True,
             'controller_nsight_options': {'cuda-graph-trace': 'graph',
                                           'cuda-memory-usage': 'true',
                                           'trace': 'cuda,nvtx,cublas,ucx'},
             'critic_warmup': 0,
             'default_hdfs_dir': None,
             'default_local_dir': '/lpai/output/models',
             'del_local_ckpt_after_load': False,
             'device': 'cuda',
             'esi_redundant_time': 0,
             'experiment_name': 'frozenlake-agent-32B-workflow',
             'log_val_generations': 0,
             'logger': ['console', 'tensorboard'],
             'max_actor_ckpt_to_keep': None,
             'max_critic_ckpt_to_keep': None,
             'n_gpus_per_node': 8,
             'nnodes': 1,
`

after 3 steps training , always generate !!!!!!!!!!!!!!!!!

<img width="1418" height="2324" alt="Image" src="https://github.com/user-attachments/assets/ecdcdc41-3b87-427d-b98b-b0ba0a16dcce" />

**Comment by yiyepiaoling0715** [comment]
use rllm
